ts/Data/Connectors/GoogleSheetsConnector.ts

```diff
@@ -133,14 +135,52 @@ class GoogleSheetsConnector extends DataConnector {
     /**
      * The attached converter, which can be replaced in the constructor
      */
-    public readonly converter: GoogleSheetsConverter;
+    public converter: GoogleSheetsConverter;
 
     /* *
      *
      *  Functions
      *
      * */
 
+    /**
+     * Iterates over the data tables and initiates the corresponding converters.
+     * Assigns the first converter.
+     *
+     * @param {GoogleSheetsConverter.GoogleSpreadsheetJSON}[json]
+     * Data retrieved from the provided URL.
+     */
+    public initConverters(
+        json: GoogleSheetsConverter.GoogleSpreadsheetJSON
+    ): void {
+        let index = 0;
+        for (const [key, table] of Object.entries(this.dataTables)) {
+            const options = this.options;
+            // Takes over the connector default options.
+            const dataTableOptions = {
+                dataTableKey: key,
+                firstRowAsNames: table.firstRowAsNames ??
+                    options.firstRowAsNames,
+                beforeParse: table.beforeParse ?? options.beforeParse
+            };
```

Comment:
Lines 157-165, are the same as in CSV Connector (and others). Can we extract it to the function in DataConnector?

Reply:
It's a bit tricky because the options for every connector type change. Also, the data we pass to the converter parse method differs (not only in type, but it's defined as `data`/`csv`/`json`). Before, I tried to move the entire method, but ended up with multiple TS errors and highly unreadable code. 

I pushed the [new version](https://github.com/highcharts/highcharts/pull/22927/commits/028fdc4da0e29db23332cd5cd92c42487e046a61), let me know what you think 😅.